presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -276,14 +277,24 @@ public void recordSubqueries(Node node, ExpressionAnalysis expressionAnalysis)
         return ImmutableList.of();
     }
 
-    public void addJoinInPredicates(Join node, JoinInPredicates joinInPredicates)
+    public void addJoinSideExpressions(Join node, JoinSideExpressions joinSideExpressions)
     {
-        this.joinInPredicates.put(node, joinInPredicates);
+        this.joinSideExpressions.put(node, joinSideExpressions);
     }
 
-    public JoinInPredicates getJoinInPredicates(Join node)
+    public JoinSideExpressions getJoinSideExpressions(Join node)
     {
-        return joinInPredicates.get(node);
+        return joinSideExpressions.get(node);
+    }
+
+    public void setPostJoinConjunct(Join node, Expression postJoinConjunct)
```

Comment:
Exactly, `postJoindPredicate` is a WHERE predicate applied just after the join.

When I was applying your comments, I realized that this `postJoinPredicate` is actually known to `RelationPlanner`. `RelationPlanner` retrieves that from regular `joinCriteria`. The same thing for JoinSideExpressions. So I removed them from `Analysis`.